src/Chapter4.hs

```diff
@@ -673,6 +714,24 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+data Tree a = Node a (Tree a) (Tree a) | Nil
+
+instance (Show a) => Show (Tree a) where
+  show Nil = ""
+  show (Node a l r) = show l ++ " " ++ show a ++ " " ++ show r
+
+instance Functor Tree where
+  fmap _ Nil = Nil
+  fmap f (Node a l r) = Node (f a) (fmap f l) (fmap f r)
+
+reverseTree:: Tree a -> Tree a
+reverseTree Nil = Nil
+reverseTree (Node a l r) = Node a r l
```

Comment:
Unfortunately, this is not enough and you also need to reverse left and right subtrees recursively 🌳